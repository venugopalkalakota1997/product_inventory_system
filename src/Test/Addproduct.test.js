import React from 'react';
import { MemoryRouter as Router, withRouter } from 'react-router-dom'
import {mount} from 'enzyme';
import {shallow} from 'enzyme';
import Addproduct from '../compoents/Addproduct';
import {configure} from 'enzyme'
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter:new Adapter()});

it('Render',()=>{
  const wrapper=shallow(<Addproduct/>);
  const h=wrapper.find("h2");
  const result=h.text();

  expect(result).toBe("Add Product");
})

test('Addproduct', () => {
    expect(true).toBeTruthy();
  });

it('Submit Fucntion',()=>{
  const onSubmitFun=jest.fn();
  const wrapper=mount(<form onSubmit={onSubmitFun}></form>)
  const form=wrapper.find('form');
  form.simulate('submit');
  expect(onSubmitFun).toHaveBeenCalledTimes(1);
});
it('productname ',()=>{
  const wrapper=mount(<input type="text" id="productname"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('text');
  expect(input.prop('id')).toEqual('productname');
})
it('productdetails ',()=>{
  const wrapper=mount(<input type="text" name="productdetails"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('text');
  expect(input.prop('name')).toEqual('productdetails');
})
it('productprice ',()=>{
  const wrapper=mount(<input type="number" id="productprice"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('number');
  expect(input.prop('id')).toEqual('productprice');
})
it('productquantity ',()=>{
  const wrapper=mount(<input type="number" id="productquantity"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('number');
  expect(input.prop('id')).toEqual('productquantity');
})
it('productimage ',()=>{
  const wrapper=mount(<input type="file" />);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('file');
  
})
it('Button ',()=>{
  const wrapper=mount(<button type="submit">Login</button>);
  const input=wrapper.find('button');
  const result=input.text();
  expect(result).toBe('Login')
  expect(input.prop('type')).toEqual('submit');
})
test('mathes Snapshot',()=>{
  const tree=renderer.create(<Router><Addproduct/></Router>).toJSON()
  expect(tree).toMatchSnapshot();
})