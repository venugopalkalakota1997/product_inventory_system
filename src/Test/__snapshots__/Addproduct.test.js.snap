// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes Snapshot 1`] = `
<form
  className="form"
>
  <h2>
    Add Product
  </h2>
  <br />
  <p>
    Product name 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </p>
  <input
    className="input"
    id="productname"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
  <p
    className="error"
  >
    
  </p>
  <p>
    Product Description 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </p>
  <textarea
    id="productdetails"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  >
     
  </textarea>
  <br />
  <p
    className="error"
  >
    
  </p>
  <p>
    Price 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </p>
  <input
    className="input"
    id="productprice"
    onBlur={[Function]}
    onChange={[Function]}
    type="number"
  />
  <p
    className="error"
  >
    
  </p>
  <p>
    Quantity 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </p>
  <input
    className="input"
    id="productquantity"
    onBlur={[Function]}
    onChange={[Function]}
    type="number"
  />
  <p
    className="error"
  >
    
  </p>
  <p>
    Categorey 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </p>
  <select
    className="input"
    id="productquantity"
    onBlur={[Function]}
    onChange={[Function]}
  >
    <option
      id="productcategory"
    >
      Select option
    </option>
    <option
      id="productcategory"
    >
      Televison
    </option>
    <option
      id="productcategory"
    >
      Mobile 
    </option>
    <option
      id="productcategory"
    >
      Furniture
    </option>
    <option
      id="productcategory"
    >
      Computer Accessories
    </option>
  </select>
  <br />
  <p
    className="error"
  >
    
  </p>
  <p>
    Select Image 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </p>
  <input
    accept="image/*"
    multiple={true}
    onBlur={[Function]}
    onChange={[Function]}
    type="file"
  />
  <br />
  <p
    className="error"
  >
    
  </p>
  <br />
  <button
    className="button"
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    Add
  </button>
</form>
`;
